File: app/src/main/java/com/example/vaccineapp/model/TimelineStage.java
package com.example.vaccineapp.model;

import com.example.vaccineapp.model_kotlin.InfoElement;

import java.util.ArrayList;

public class TimelineStage {

    private ArrayList<InfoElement> timelineGroup;

    public TimelineStage(){
        this.timelineGroup = new ArrayList<>();
    }

    public int getSize(){
        return this.timelineGroup.size();
    }

    public void addInfoElement(InfoElement info){
        this.timelineGroup.add(info);
    }

    public InfoElement getInfoElement(int position){
        return this.timelineGroup.get(position);
    }
}

File: app/src/main/java/com/example/vaccineapp/model_kotlin/TimelineStage.kt
package com.example.vaccineapp.model_kotlin

class TimelineStage {

    private val timelineGroup: ArrayList<InfoElement> = arrayListOf()

    fun addInfoElement(info: InfoElement){
        timelineGroup.add(info)
    }

    fun getInfoElement(position: Int): InfoElement = timelineGroup.get(position)

    val size: Int
    get() = timelineGroup.size
}